File: source/clibCoreCs/Controllers/bak/stringBuilderLegacyController.cs



namespace Controllers {
    
    public class stringBuilderLegacyController {
        
        // 
        private int iSize;
        
        private int iCount;
        
        private string[] Holder;
        
        // 
        // ==========================================================================================
        // '' <summary>
        // '' add a string to the stringbuilder
        // '' </summary>
        // '' <param name="NewString"></param>
        public void Add(string NewString) {
            try {
                if ((iCount >= iSize)) {
                    iSize = (iSize + iChunk);
                    object Preserve;
                    Holder[iSize];
                }
                
                Holder[iCount] = NewString;
                iCount = (iCount + 1);
            }
            catch (Exception ex) {
                throw new ApplicationException("Exception in coreFastString.Add()", ex);
            }
            
        }
        
        // 
        // ==========================================================================================
        // '' <summary>
        // '' read the string out of the string builder
        // '' </summary>
        // '' <returns></returns>
        public string Text {
            get {
                return (Join(Holder, "") + "");
            }
        }
    }
}

File: source/Processor/Controllers/22/stringBuilderLegacyController.vb

Option Explicit On
Option Strict On

Namespace Contensive.Core.Controllers
    Public Class stringBuilderLegacyController
        '
        Private iSize As Integer
        Private Const iChunk = 100
        Private iCount As Integer
        Private Holder() As String
        '
        '==========================================================================================
        ''' <summary>
        ''' add a string to the stringbuilder
        ''' </summary>
        ''' <param name="NewString"></param>
        Public Sub Add(ByVal NewString As String)
            Try
                If iCount >= iSize Then
                    iSize = iSize + iChunk
                    ReDim Preserve Holder(iSize)
                End If
                Holder(iCount) = NewString
                iCount = iCount + 1
            Catch ex As Exception
                Throw New ApplicationException("Exception in coreFastString.Add()", ex)
            End Try
        End Sub
        '
        '==========================================================================================
        ''' <summary>
        ''' read the string out of the string builder
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property Text As String
            Get
                Return Join(Holder, "") & ""
            End Get
        End Property
    End Class
End Namespace